test/ResolutionManager.ts

```diff
@@ -1610,4 +1570,97 @@ describe("Resolution", async () => {
       );
     });
   });
+
+  describe("distrust resolution", async () => {
+    let totalVotingPower = 100;
+    async function _prepare() {
+      await resolution
+        .connect(managingBoard)
+        .createDistrustResolution("test", 6, false, [], [], user2.address);
+
+      voting.getTotalVotingPowerAt
+        .whenCalledWith(resolutionSnapshotId)
+        .returns(totalVotingPower);
+
+      await resolution.connect(managingBoard).approveResolution(resolutionId);
+      const approveTimestamp = await getEVMTimestamp();
+      await setEVMTimestamp(approveTimestamp + 3 * DAY);
+    }
+
+    it("should not remove and re-add delegation when not delegating", async () => {
+      await resolution
+        .connect(managingBoard)
+        .createDistrustResolution("test", 0, false, [], [], user2.address);
+
+      voting.getDelegate.whenCalledWith(user2.address).returns(user2.address);
+
+      await resolution.connect(managingBoard).approveResolution(resolutionId);
+
+      expect(voting.delegateOnBehalf).to.have.not.been.called;
+    });
+
+    it("should remove and re-add delegation when delegating", async () => {
+      await resolution
+        .connect(managingBoard)
+        .createDistrustResolution("test", 0, false, [], [], user2.address);
+
+      voting.getDelegate.whenCalledWith(user2.address).returns(user1.address);
+
+      await resolution.connect(managingBoard).approveResolution(resolutionId);
+
+      expect(voting.delegateOnBehalf.getCall(0).args).eql([
+        user2.address,
+        user2.address,
+      ]);
+      expect(voting.delegateOnBehalf.getCall(1).args).eql([
+        user2.address,
+        user1.address,
+      ]);
+    });
+
+    it("should prevent distrusted from voting", async () => {
+      await _prepare();
+      setupUser(user2, 0);
+
+      await expect(
+        resolution.connect(user2).vote(resolutionId, false)
+      ).revertedWith("Resolution: account cannot vote");
+    });
+
+    it("should not count distrusted balance for quorum", async () => {
```

Comment:
Does this implicitly checks the value (as a number) of the quorum as well?

Reply:
implicity I suppose yes. if the resolution passes, it means the quorum has been reached